src/main/cpp/src/cast_string_to_timestamp.cu

```diff
@@ -655,27 +655,29 @@ __device__ RESULT_TYPE parse_timestamp_string(unsigned char const* const ptr,
 
   if (eof(pos, end_pos)) { return RESULT_TYPE::INVALID; }
 
-  // parse sign
-  bool negative_year_sign    = false;
-  unsigned char const sign_c = ptr[pos];
-  if ('-' == sign_c || '+' == sign_c) {
-    pos++;
-    if ('-' == sign_c) { negative_year_sign = true; }
-  }
-
-  if (eof(pos, end_pos)) { return RESULT_TYPE::INVALID; }
-
   ts_segments ts;
+  bool negative_year_sign = false;
 
-  if (ptr[pos] == 'T') {
-    // after parse sign, first char is T, means only have time part
+  if (ptr[pos] == 'T' || (end_pos - pos > 1 && ptr[pos + 1] == ':') ||
+      (end_pos - pos > 2 && (ptr[pos + 1] == ':' || ptr[pos + 2] == ':'))) {
```

Comment:
question: Is "00" valid? Seeing that `T00` is valid in test

Reply:
00 is invalid, T00 is valid.
```
>>> spark.sql("select cast('00' as timestamp)").show()
+---------------------+
|CAST(00 AS TIMESTAMP)|
+---------------------+
|                 NULL|
+---------------------+

>>> spark.sql("select cast('T00' as timestamp)").show()
+----------------------+
|CAST(T00 AS TIMESTAMP)|
+----------------------+
|   2025-05-21 00:00:00|
+----------------------+
```

And `0001` is also valid, it's 0001-01-01 date.